akka-http-core/src/main/scala/akka/http/scaladsl/Http.scala

```diff
@@ -360,7 +364,9 @@ class HttpExt private[http] (private val config: Config)(implicit val system: Ex
     val server = HttpServerBluePrint(settings, log, isSecureConnection)
       .addAttributes(HttpAttributes.remoteAddress(remoteAddress))
 
-    server atop delayCancellationStage(settings)
+    server
+      .atop(delayCancellationStage(settings)) // TODO: figure out if we still need that at all
+      .addAttributes(cancellationStrategyAttributeForDelay(settings.streamCancellationDelay))
```

Comment:
Yeah, I would have guessed the delay cancellation stage becomes no longer needed with the delay built in